Sources/FuturedHelpers/Helpers/CoordinatorSceneFlowProvider.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  CoordinatorSceneFlowProvider.swift
+//
+//  Created by Simon Sestak on 31/07/2024.
+//
+
+import SwiftUI
+import FuturedArchitecture
+
+/**
+ A typealias representing navigable destinations in a reusable scene flow.
+
+ # Notes: #
+ 1. If the scenes defined in the provider can continue with other scenes, define an end destination.
+    - This end destination should trigger a scene display outside of the scene provider.
+ 2. Other destinations will be used to display scenes defined within the scene provider.
+
+ # Example #
+ ```
+ protocol TemplateFlowDestination: CoordinatorSceneFlowDestination {
+    static var destination: Self { get }
+    static var end: Self { get }
```

Comment:
We probably should use naming which does not appear to have a special meaning, if it is only an example.

```suggestion
    static var someDestination: Self { get }
    static var otherDestionation: Self { get }
```